### DISCUSSION TITLE
GraphQL Lodash Functions

### DISCUSSION THREAD

**Discussion by sksar** [directus/directus#17824] [Feature Requests]
### Summary

There could be a lot more power unlocked from Directus if the GraphQL API supported many advanced features.
An example would be to do lodash-like processing directly in GraphQL.

Here is a library that integrates it with existing GraphQL Schemas.
https://github.com/IvanGoncharov/graphql-lodash

This way, functions from lodash can be used to modify the data received directly from the server.
It would be much beneficial in use cases where there could be complex manipulations of the data that needed to be performed on the client side.

### Motivation

Adding this to the directus GraphQL API would be beneficial for a lot of developers.
This way, we can do so much stuff and processing of data, directly in the GraphQL.
This will unlock a lot of potential for the GraphQL API.

### Detailed Design

Adding this to the directus Graphql API would be beneficial for a lot of developers.

### Should Have:
- Integrating this library: https://github.com/IvanGoncharov/graphql-lodash

### Could Have:
- Other libraries similar to the above

### Drawbacks

Not sure,
But there is a possibility this could not work well with the Permissions layer.
Not sure, how this affects the REST API.

### Alternatives

Not sure

### Adoption Strategy

Not sure

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
